android/src/com/mapswithme/maps/editor/MultilanguageAdapter.java

```diff
@@ -64,13 +76,22 @@ public int getItemCount()
     return mNames.size();
   }
 
-  public static class Holder extends RecyclerView.ViewHolder
+  public class Holder extends RecyclerView.ViewHolder
   {
     EditText input;
 
     public Holder(View itemView)
     {
       super(itemView);
+      input = (EditText) itemView.findViewById(R.id.input);
+      itemView.findViewById(R.id.delete).setOnClickListener(new View.OnClickListener()
+                                                            {
+                                                              @Override
+                                                              public void onClick(View v)
+                                                              {
+                                                                mNames.removeItemAt(getAdapterPosition());
```

Comment:
Ага, это фишка SortedList такая, там ссылка на адаптер внутри.
> On Apr 28, 2016, at 12:38 PM, Alexander Marchuk <notifications@github.com> wrote:
> 
> In android/src/com/mapswithme/maps/editor/MultilanguageAdapter.java <https://github.com/mapsme/omim/pull/3042#discussion_r61399445>:
> 
> >    {
> >      EditText input;
> >  
> >      public Holder(View itemView)
> >      {
> >        super(itemView);
> > +      input = (EditText) itemView.findViewById(R.id.input);
> > +      itemView.findViewById(R.id.delete).setOnClickListener(new View.OnClickListener()
> > +                                                            {
> > +                                                              @Override
> > +                                                              public void onClick(View v)
> > +                                                              {
> > +                                                                mNames.removeItemAt(getAdapterPosition());
> А разве список сам обновится после удаления?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/3042/files/5a529e4b26be1b1292774830b96dbfaed83817c8..a8256f894529cb51f3b8f066d44706a6f9aac729#r61399445>